tfsdk/attribute_plan_modification_test.go

```diff
@@ -0,0 +1,682 @@
+package tfsdk
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+func TestRequiresReplaceModifier(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		state        State
+		plan         Plan
+		config       Config
+		path         *tftypes.AttributePath
+		expectedPlan attr.Value
+		expectedRR   bool
+	}
+
+	schema := Schema{
+		Attributes: map[string]Attribute{
+			"a": {
```

Comment:
Nit: It could be clearer down in the test cases if these names are more descriptive, e.g. `optional-computed` and `optional`